gns/train.py

```diff
@@ -246,24 +248,24 @@ def train(rank, flags, world_size):
   not_reached_nsteps = True
   try:
     while not_reached_nsteps:
-      if type(rank) == int:
+      if device == torch.device("cuda"):
         torch.distributed.barrier()
       else:
         pass
       for ((position, particle_type, n_particles_per_example), labels) in dl:
-        position.to(rank)
-        particle_type.to(rank)
-        n_particles_per_example.to(rank)
-        labels.to(rank)
+        position.to(rank if device == torch.device("cuda") else device)
+        particle_type.to(rank if device == torch.device("cuda") else device)
+        n_particles_per_example.to(rank if device == torch.device("cuda") else device)
+        labels.to(rank if device == torch.device("cuda") else device)
 
         # TODO (jpv): Move noise addition to data_loader
         # Sample the noise to add to the inputs to the model during training.
-        sampled_noise = noise_utils.get_random_walk_noise_for_position_sequence(position, noise_std_last_step=flags["noise_std"]).to(rank)
-        non_kinematic_mask = (particle_type != KINEMATIC_PARTICLE_ID).clone().detach().to(rank)
+        sampled_noise = noise_utils.get_random_walk_noise_for_position_sequence(position, noise_std_last_step=flags["noise_std"]).to(rank if device == torch.device("cuda") else device)
+        non_kinematic_mask = (particle_type != KINEMATIC_PARTICLE_ID).clone().detach().to(rank if device == torch.device("cuda") else device)
```

Comment:
To avoid repeating, the best way I can come up with is to predefine the rank as something like the following, since the rank takes integer if cuda else `torch.device('cpu')`
```
if torch.device('cpu'):
  rank = torch.device("cpu")
else:
  rank = 0   # If the primary device need to gather gradients.
```